src/wp-includes/class-wp-autoload.php

```diff
@@ -0,0 +1,500 @@
+		'wp_terms_list_table'                         => 'wp-admin/includes/class-wp-terms-list-table.php',
+		'wp_theme_install_list_table'                 => 'wp-admin/includes/class-wp-theme-install-list-table.php',
+		'wp_themes_list_table'                        => 'wp-admin/includes/class-wp-themes-list-table.php',
+		'wp_upgrader_skin'                            => 'wp-admin/includes/class-wp-upgrader-skin.php',
+		'wp_upgrader'                                 => 'wp-admin/includes/class-wp-upgrader.php',
+		'wp_users_list_table'                         => 'wp-admin/includes/class-wp-users-list-table.php',
+
+		/* Classes in wp-admin/includes/deprecated.php. */
+		'wp_user_search'                              => 'wp-admin/includes/deprecated.php',
+		'wp_privacy_data_export_requests_table'       => 'wp-admin/includes/deprecated.php',
+		'wp_privacy_data_removal_requests_table'      => 'wp-admin/includes/deprecated.php',
+	);
+
+	/**
+	 * Whether the autoloader has already been registered or not.
+	 *
+	 * Avoid registering the autoloader multiple times.
+	 *
+	 * @static
+	 * @access private
+	 *
+	 * @var bool
+	 */
+	private static $registered = false;
+
+	/**
+	 * Register the autoloader.
+	 *
+	 * @return void
+	 */
+	public static function register() {
+		// Bail early if already registered.
+		if ( self::$registered ) {
+			return;
+		}
+
+		self::register_external_bundled();
+		self::register_core();
+
+		self::$registered = true;
+	}
+
+	/**
+	 * Register the autoloader for external, bundled libraries.
+	 *
+	 * @return void
+	 */
+	public static function register_external_bundled() {
+		require_once ABSPATH . 'wp-includes/Requests/src/Autoload.php';
+		require_once ABSPATH . 'wp-includes/sodium_compat/autoload.php';
+
+		spl_autoload_register( array( '\WpOrg\Requests\Autoload', 'load' ) );
+	}
+
+	/**
+	 * Register the autoloader for WordPress Core classes.
+	 *
+	 * @return void
+	 */
+	public static function register_core() {
+		spl_autoload_register( array( __CLASS__, 'autoload_core' ), true, true );
+	}
+
+	/**
+	 * Autoload a WordPress class.
+	 *
+	 * @param string $class_name Class name.
+	 * @return void
+	 */
+	public static function autoload_core( string $class_name ) {
+		// Lowercase the classname to accommodate for WP classes written with wrong cases.
+		$class_name = strtolower( $class_name );
+
+		// Load Avifinfo classes.
+		if ( str_starts_with( $class_name, 'avifinfo' ) ) {
+			// This file contains multiple classes, so we need to use require_once.
+			require_once ABSPATH . 'wp-includes/class-avif-info.php';
+			return;
+		}
+
+		// Load SimplePie classes.
+		if ( str_starts_with( $class_name, 'simplepie' ) ) {
+			require_once ABSPATH . 'wp-includes/class-simplepie.php';
+			return;
+		}
+
+		// Bail early if the class is not a WP class.
+		// Use empty() instead of !isset() for performance reasons (saves a BOOL_NOT opcode).
```

Comment:
Coding standards for [multi-line comments](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#5-inline-comments) 

```suggestion
		/*
		 * Bail early if the class is not a WP class.
		 * Use empty() instead of !isset() for performance reasons (saves a BOOL_NOT opcode).
		 */
```